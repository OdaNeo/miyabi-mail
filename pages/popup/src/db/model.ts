import { PROMPT_KEYS } from '@/utils/tts';
import Dexie, { type EntityTable } from 'dexie';

interface TaskHistory {
  id: number;
  sourceContent: string;
  targetLanguage: string;
  taskType: PROMPT_KEYS;
  modelVersion: string;
  modelTemperature: number;
  emailSubject?: string;
  generatedContent: string;
  createdAt: Date;
  completedAt: Date;
  isVisitable: boolean;
}

const historyDb = new Dexie('MiyabiMailHistory') as Dexie & {
  taskRecords: EntityTable<TaskHistory, 'id'>;
};

historyDb.version(1).stores({
  taskRecords: [
    '++id',
    'createdAt',
    'sourceContent',
    'targetLanguage',
    'taskType',
    'modelVersion',
    'modelTemperature',
    'emailSubject',
    'generatedContent',
    'completedAt',
    'isVisitable',
  ].join(','),
});

export type { TaskHistory };
export { historyDb };
