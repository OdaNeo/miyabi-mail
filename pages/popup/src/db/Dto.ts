import { PROMPT_KEYS } from '@/utils/tts';
import { TaskHistory } from './model';

export interface TaskHistoryDTO {
  inputText: string;
  language: string;
  promptKey: PROMPT_KEYS;
  apiVersion: string;
  temperature: number;
  subject?: string;
  result: string;
  createTime: string;
  completedTime: string;
}

export class TaskHistoryMapper {
  static toEntity(dto: TaskHistoryDTO): Omit<TaskHistory, 'id'> {
    return {
      sourceContent: dto.inputText,
      targetLanguage: dto.language,
      taskType: dto.promptKey,
      modelVersion: dto.apiVersion,
      modelTemperature: dto.temperature,
      emailSubject: dto?.subject || undefined,
      generatedContent: dto.result,
      createdAt: new Date(dto.createTime),
      completedAt: new Date(dto.completedTime),
      isVisitable: true,
    };
  }
  static toViewModel(entity: TaskHistory): TaskHistoryDTO {
    return {
      inputText: entity.sourceContent,
      language: entity.targetLanguage,
      promptKey: entity.taskType,
      apiVersion: entity.modelVersion,
      temperature: entity.modelTemperature,
      subject: entity?.emailSubject || undefined,
      result: entity.generatedContent,
      createTime: entity.createdAt.toLocaleString(),
      completedTime: entity.completedAt.toLocaleString(),
    };
  }
}
