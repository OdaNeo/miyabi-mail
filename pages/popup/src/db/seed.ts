import { TaskHistory } from './model';

export const seed: Omit<TaskHistory, 'id'>[] = [
  {
    sourceContent: 'Hello, how are you?',
    targetLanguage: 'ja',
    taskType: 'POLISHING',
    modelVersion: 'GPT-3.5-turbo',
    modelTemperature: 0.7,
    generatedContent: 'fine thank you',
    createdAt: new Date(Date.now() - 1000 * 60 * 60 * 24 * 3),
    completedAt: new Date(Date.now() - 1000 * 60 * 60 * 24 * 2),
    isVisitable: true,
  },
  {
    sourceContent: 'nihaoma',
    targetLanguage: 'ja',
    taskType: 'TRANSLATION',
    modelVersion: 'GPT-4.0',
    modelTemperature: 0.6,
    generatedContent: 'こんにちは',
    createdAt: new Date(Date.now() - 1000 * 60 * 60 * 24 * 9),
    completedAt: new Date(Date.now() - 1000 * 60 * 60 * 24 * 7),
    isVisitable: true,
  },
  {
    sourceContent: 'Good morning, have a nice day!',
    targetLanguage: 'fr',
    taskType: 'TRANSLATION',
    modelVersion: 'GPT-3.5-turbo',
    modelTemperature: 0.5,
    generatedContent: 'Bonjour, passez une bonne journée!',
    createdAt: new Date(Date.now() - 1000 * 60 * 60 * 24 * 5),
    completedAt: new Date(Date.now() - 1000 * 60 * 60 * 24 * 4),
    isVisitable: true,
  },
  {
    sourceContent: 'The quick brown fox jumps over the lazy dog.',
    targetLanguage: 'de',
    taskType: 'POLISHING',
    modelVersion: 'GPT-4.0',
    modelTemperature: 0.8,
    generatedContent: 'Der schnelle braune Fuchs springt über den faulen Hund.',
    createdAt: new Date(Date.now() - 1000 * 60 * 60 * 24 * 10),
    completedAt: new Date(Date.now() - 1000 * 60 * 60 * 24 * 9),
    isVisitable: false,
  },
  {
    sourceContent: 'Can you help me with this code snippet?',
    targetLanguage: 'es',
    taskType: 'TRANSLATION',
    modelVersion: 'GPT-3.5-turbo',
    modelTemperature: 0.65,
    generatedContent: '¿Puedes ayudarme con este fragmento de código?',
    createdAt: new Date(Date.now() - 1000 * 60 * 60 * 24 * 6),
    completedAt: new Date(Date.now() - 1000 * 60 * 60 * 24 * 5),
    isVisitable: true,
  },
  {
    sourceContent: 'I love programming!',
    targetLanguage: 'it',
    taskType: 'POLISHING',
    modelVersion: 'GPT-4.0',
    modelTemperature: 0.7,
    generatedContent: 'Amo programmare!',
    createdAt: new Date(Date.now() - 1000 * 60 * 60 * 24 * 2),
    completedAt: new Date(Date.now() - 1000 * 60 * 60 * 24 * 1),
    isVisitable: false,
  },
];
