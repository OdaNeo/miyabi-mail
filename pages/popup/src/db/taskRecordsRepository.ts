import { historyDb, TaskHistory } from './model';

async function getAllTaskRecords() {
  return await historyDb.taskRecords.toArray();
}

async function findTaskRecordById(id: number) {
  return await historyDb.taskRecords.get(id);
}

async function deleteTaskRecordById(id: number) {
  updateTaskRecordById(id, { isVisitable: false });
}

async function updateTaskRecordById(id: number, taskRecord: Partial<Omit<TaskHistory, 'id'>>) {
  await historyDb.taskRecords.update(id, taskRecord);
}

async function addTaskRecord(taskRecord: Omit<TaskHistory, 'id'>) {
  await historyDb.taskRecords.add(taskRecord);
}

export { getAllTaskRecords, findTaskRecordById, deleteTaskRecordById, updateTaskRecordById, addTaskRecord };
