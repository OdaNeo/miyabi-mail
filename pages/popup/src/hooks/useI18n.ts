import { i18nStorage } from '@extension/storage';
import { useStorage } from '@extension/shared';
import { useEffect, useState } from 'react';

export enum I18n {
  ORIGINAL_TRANSLATION = 'ORIGINAL_TRANSLATION',
  REPLY = 'REPLY',
  COPY = 'COPY',
  COPIED = 'COPIED',
  PASTE_API_KEY = 'PASTE_API_KEY',
  IS_NOT_MAIL_CONTEST = 'IS_NOT_MAIL_CONTEST',
  GENERATE_SUBJECT = 'GENERATE_SUBJECT',
  INPUT_PLACEHOLDER = 'INPUT_PLACEHOLDER',
  IS_POLISHING = 'IS_POLISHING',
  POLISH = 'POLISH',
  TRANSLATION = 'TRANSLATION',
  IS_TRANSLATING = 'IS_TRANSLATING',
  IS_GENERATING = 'IS_GENERATING',
  GENERATE = 'GENERATE',
  INPUT_SUBJECT = 'INPUT_SUBJECT',
  CHANGE_TO_LIGHT = 'CHANGE_TO_LIGHT',
  CHANGE_TO_DARK = 'CHANGE_TO_DARK',
  API_KEY_SETTING = 'API_KEY_SETTING',
  PLEASE_SET_API_KEY = 'PLEASE_SET_API_KEY',
  SAVE = 'SAVE',
  CHANGE_LANGUAGE = 'CHANGE_LANGUAGE',
}

export const useI18n = () => {
  const lang = useStorage(i18nStorage);

  const [translations, setTranslations] = useState<Record<string, string>>({});

  useEffect(() => {
    import(`../i18n/${lang.value}.json`)
      .then(module => setTranslations(module.default))
      .catch(() => import('../i18n/cn.json').then(fallback => setTranslations(fallback.default)));
  }, [lang.value]);

  return translations;
};
