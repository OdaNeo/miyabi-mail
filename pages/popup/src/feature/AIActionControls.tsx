import { Button, Progress, Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@extension/ui';
import { Languages, Wand2Icon } from 'lucide-react';
import { useI18n } from '../hooks/useI18n';
import { rolePrompt } from '../utils/tts';
import { useProgress } from '../hooks/useProgress';
import { useGeneratedStore } from '../store/generatedStore';
import { useExpandedSectionStore } from '../store/expandedSectionStore';
import { languageStorage, replyStorage, translationStorage } from '@extension/storage';
import { useReplyLoading } from '@/store/replyLoadingStore';
import { useTranslationLoading } from '@/store/translationLoadingStore';
import { usePolishingLoading } from '@/store/polishingLoadingStore';
import { useStorage } from '@extension/shared';
import { LANGUAGES } from '@/const/language';

export function AIActionControls({
  inputText,
  subject,
  setInputText,
  runOpenAIAction,
  handleOnClick,
}: {
  inputText: string;
  subject: string;
  setInputText: (text: string) => void;
  runOpenAIAction: ({
    prompt,
    onStart,
    onFinish,
    onSuccess,
  }: {
    prompt: string;
    onStart?: () => void;
    onFinish?: () => void;
    onSuccess?: (content: string) => void;
  }) => Promise<void>;
  handleOnClick: () => void;
}) {
  const { IS_POLISHING, POLISH, IS_TRANSLATING, TRANSLATION, IS_GENERATING, GENERATE } = useI18n();

  const language = useStorage(languageStorage);

  const { setIsGenerated } = useGeneratedStore();
  const { setExpandedSection } = useExpandedSectionStore();
  const { isReplyLoading, setIsReplyLoading } = useReplyLoading();
  const { isTranslationLoading, setIsTranslationLoading } = useTranslationLoading();
  const { isPolishingLoading, setIsPolishingLoading } = usePolishingLoading();

  const [replyProgress, setReplyProgress] = useProgress(isReplyLoading);
  const [translationProgress, setTranslationProgress] = useProgress(isTranslationLoading);
  const [polishingProgress, setPolishingProgress] = useProgress(isPolishingLoading);

  const isLoading = isReplyLoading || isTranslationLoading || isPolishingLoading;

  const handleReply = () => {
    runOpenAIAction({
      prompt: rolePrompt(inputText?.trim(), 'REPLY', language, subject?.trim()),
      onStart: () => {
        setIsGenerated(false), setIsReplyLoading(true), setReplyProgress(0);
      },
      onFinish: () => {
        setIsReplyLoading(false), setReplyProgress(0);
      },
      onSuccess: content => {
        setIsGenerated(true), replyStorage.set(content), setExpandedSection('REPLY');
      },
    });
  };

  const handleTranslation = () => {
    runOpenAIAction({
      prompt: rolePrompt(inputText?.trim(), 'TRANSLATION', language),
      onStart: () => {
        setIsGenerated(false), setIsTranslationLoading(true), setTranslationProgress(0);
      },
      onFinish: () => {
        setIsTranslationLoading(false), setTranslationProgress(0);
      },
      onSuccess: content => {
        setIsGenerated(true), translationStorage.set(content), setExpandedSection('TRANSLATION');
      },
    });
  };

  const handlePolishing = () => {
    runOpenAIAction({
      prompt: rolePrompt(inputText?.trim(), 'POLISHING'),
      onStart: () => {
        setIsGenerated(false), setIsPolishingLoading(true), setPolishingProgress(0);
      },
      onFinish: () => {
        setIsPolishingLoading(false), setPolishingProgress(0);
      },
      onSuccess: content => {
        setIsGenerated(false), setInputText(content), setExpandedSection(null);
      },
    });
  };

  return (
    <div className="flex items-center" onClick={handleOnClick}>
      <Select value={language} onValueChange={languageStorage.set}>
        <SelectTrigger className="w-32 h-8 bg-slate-100 dark:bg-slate-800 text-slate-800 dark:text-slate-200">
          <SelectValue placeholder="选择语言" />
        </SelectTrigger>
        <SelectContent>
          {LANGUAGES.map(lang => (
            <SelectItem key={lang.value} value={lang.value} className="text-sm">
              {lang.label}
            </SelectItem>
          ))}
        </SelectContent>
      </Select>
      <div className="flex-1" />
      <Button
        onClick={handlePolishing}
        disabled={isLoading || !inputText?.trim()}
        className="relative overflow-hidden ml-2 h-8 px-3 text-ms
          bg-slate-200 hover:bg-slate-300 text-slate-900
          dark:bg-slate-700 dark:hover:bg-slate-600 dark:text-slate-200"
      >
        <Wand2Icon className="h-4 w-4" />
        {isPolishingLoading ? <span>{IS_POLISHING}</span> : <span>{POLISH}</span>}
        {isPolishingLoading && (
          <Progress
            value={polishingProgress}
            className="absolute bottom-0 left-0 right-0 h-1 bg-emerald-400 dark:bg-emerald-500"
          />
        )}
      </Button>
      <Button
        onClick={handleTranslation}
        disabled={isLoading || !inputText?.trim()}
        className="relative overflow-hidden ml-2 h-8 px-3 text-ms
          bg-slate-200 hover:bg-slate-300 text-slate-900
          dark:bg-slate-700 dark:hover:bg-slate-600 dark:text-slate-200"
      >
        <Languages className="h-4 w-4" />
        {isTranslationLoading ? <span>{IS_TRANSLATING}</span> : <span>{TRANSLATION}</span>}
        {isTranslationLoading && (
          <Progress
            value={translationProgress}
            className="absolute bottom-0 left-0 right-0 h-1 bg-emerald-400 dark:bg-emerald-500"
          />
        )}
      </Button>
      <Button
        onClick={handleReply}
        data-testid="reply-button"
        disabled={isLoading || !inputText?.trim()}
        className="relative overflow-hidden ml-2 h-8 px-3 text-ms
          bg-emerald-500 hover:bg-emerald-600 text-white
          dark:bg-emerald-600 dark:hover:bg-emerald-700"
      >
        {isReplyLoading ? <span>{IS_GENERATING}</span> : <span>{GENERATE}</span>}
        {isReplyLoading && (
          <Progress
            value={replyProgress}
            className="absolute bottom-0 left-0 right-0 h-1 bg-emerald-300 dark:bg-emerald-400"
          />
        )}
      </Button>
    </div>
  );
}
