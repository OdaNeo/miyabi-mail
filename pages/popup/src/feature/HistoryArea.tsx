import { MessageBox } from '@/components/MessageBox';
import { historyDb } from '@/db/model';
import { deleteTaskRecordById } from '@/db/taskRecordsRepository';
import { useI18n } from '@/hooks/useI18n';
import { taskTypeConfig } from '@/utils/taskTypeConfig';
import { Button, ScrollArea } from '@extension/ui';
import { useLiveQuery } from 'dexie-react-hooks';
import { AnimatePresence, motion } from 'framer-motion';
import { Clock, Trash2, AlertCircle } from 'lucide-react';

export const HistoryArea = () => {
  const filteredHistory = useLiveQuery(() =>
    historyDb.taskRecords
      .filter(item => item.isVisitable)
      .reverse()
      .sortBy('completedAt'),
  );

  const { NO_HISTORY, HISTORY_IS_HERE } = useI18n();

  const handleDeleteHistory = async (id: number) => {
    await deleteTaskRecordById(id);
  };

  return (
    <ScrollArea className="scrollbar-custom p-5">
      <div className="space-y-3">
        <AnimatePresence>
          {filteredHistory && filteredHistory.length > 0 ? (
            filteredHistory.map(item => (
              <motion.div
                key={item.id}
                data-testid="history-item"
                initial={{ opacity: 0, y: 10 }}
                animate={{ opacity: 1, y: 0 }}
                exit={{ opacity: 0, height: 0, marginTop: 0, marginBottom: 0, overflow: 'hidden' }}
                transition={{ duration: 0.2 }}
                className="border border-[#e5e7eb] dark:border-[#2e3238] rounded-lg overflow-hidden transition-all duration-200"
              >
                <div className="flex flex-col">
                  <div
                    className="flex justify-between items-center p-3 border-b px-3 h-9 cursor-pointer
                     bg-slate-100 dark:bg-[#23252b]
                 hover:bg-slate-200 dark:hover:bg-slate-800"
                  >
                    <div className="flex items-center gap-2 text-[#6b7280] dark:text-[#9ca3af]">
                      <Clock className="w-3.5 h-3.5" />
                      <span>{item.completedAt.toLocaleString()}</span>
                    </div>
                    <div className="flex items-center gap-2">
                      <span
                        className={`px-2 py-0.5 rounded-full ${taskTypeConfig[item.taskType]?.color || 'bg-gray-100 text-gray-800 dark:bg-gray-700 dark:text-gray-300'}`}
                      >
                        {taskTypeConfig[item.taskType]?.label || item.taskType}
                      </span>
                      <Button
                        variant="ghost"
                        size="icon"
                        data-testid="delete-history-icon"
                        onClick={() => handleDeleteHistory(item.id)}
                        className="h-6 w-6 p-1.5 rounded-full text-[#9ca3af] hover:text-[#ef4444] dark:text-[#6b7280] dark:hover:text-[#f87171] hover:bg-[#fee2e2]/30 dark:hover:bg-[#450a0a]/20 transition-colors"
                      >
                        <Trash2 className="w-3.5 h-3.5" />
                      </Button>
                    </div>
                  </div>
                  <MessageBox>{item.sourceContent}</MessageBox>
                  <div className="border-b" />
                  <MessageBox>{item.generatedContent}</MessageBox>
                </div>
              </motion.div>
            ))
          ) : (
            <motion.div
              initial={{ opacity: 0 }}
              animate={{ opacity: 1 }}
              className="flex flex-col items-center justify-center py-10 text-[#6b7280] dark:text-[#9ca3af]"
            >
              <AlertCircle className="w-10 h-10 mb-3 opacity-50" />
              <p className="text-sm">{NO_HISTORY}</p>
              <p className="text-sm mt-1 max-w-xs text-center">{HISTORY_IS_HERE}</p>
            </motion.div>
          )}
        </AnimatePresence>
      </div>
    </ScrollArea>
  );
};
