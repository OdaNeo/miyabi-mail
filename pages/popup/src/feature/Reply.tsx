import type { PROMPT_KEYS } from '@/utils/tts';
import { AnimatePresence, motion } from 'framer-motion';
import { Label, Button, cn } from '@extension/ui';
import { ChevronDown, ChevronUp, ClipboardCopy } from 'lucide-react';
import { useState } from 'react';
import { MessageBox } from '../components/MessageBox';
import { replyStorage, translationStorage } from '@extension/storage';
import { useStorage } from '@extension/shared';
import { useGeneratedStore } from '@/store/generatedStore';
import { useExpandedSectionStore } from '@/store/expandedSectionStore';
import { useI18n } from '@/hooks/useI18n';

export function Reply() {
  const reply = useStorage(replyStorage);
  const translation = useStorage(translationStorage);

  const { isGenerated } = useGeneratedStore();
  const { expandedSection, setExpandedSection } = useExpandedSectionStore();

  const [copied, setCopied] = useState(false);
  const { ORIGINAL_TRANSLATION, REPLY, COPY, COPIED } = useI18n();

  const toggleSection = (section: PROMPT_KEYS) => {
    setExpandedSection(expandedSection === section ? null : section);
  };

  const handleCopy = () => {
    navigator.clipboard.writeText(reply).then(() => {
      setCopied(true);
      setTimeout(() => setCopied(false), 2000);
    });
  };

  return (
    <AnimatePresence>
      {isGenerated && (
        <motion.div
          initial={{ opacity: 0, height: 0 }}
          animate={{ opacity: 1, height: 'auto' }}
          exit={{ opacity: 0, height: 0 }}
          className="space-y-2 overflow-y-auto scrollbar-custom"
        >
          {(['TRANSLATION', 'REPLY'] as PROMPT_KEYS[]).map(section => (
            <div key={section} className="space-y-1">
              <div
                className="flex justify-between items-center cursor-pointer h-8 p-1 rounded hover:bg-slate-100 dark:hover:bg-slate-800 transition-colors"
                onClick={() => toggleSection(section)}
                onKeyUp={e => {
                  if (e.key === 'Enter' || e.key === ' ') {
                    toggleSection(section);
                  }
                }}
                role="button"
                tabIndex={0}
              >
                <Label className="font-medium text-slate-700 dark:text-slate-300">
                  {section === 'TRANSLATION' ? ORIGINAL_TRANSLATION : REPLY}
                </Label>
                <div className="flex items-center space-x-1">
                  {section === 'REPLY' && (
                    <Button
                      variant="outline"
                      size="sm"
                      data-testid="copy-button"
                      onClick={e => {
                        e.stopPropagation();
                        handleCopy();
                      }}
                      className={cn(
                        'h-6 text-sm transition-colors',
                        copied
                          ? 'bg-emerald-500 text-white hover:bg-emerald-600'
                          : 'text-slate-600 dark:text-slate-400',
                      )}
                    >
                      {copied ? COPIED : COPY}
                      <ClipboardCopy className="h-3 w-3 ml-1" />
                    </Button>
                  )}
                  {expandedSection === section ? (
                    <ChevronUp className="h-3 w-3 text-slate-500 dark:text-slate-400" />
                  ) : (
                    <ChevronDown className="h-3 w-3 text-slate-500 dark:text-slate-400" />
                  )}
                </div>
              </div>
              {expandedSection === section && (
                <MessageBox>{section === 'TRANSLATION' ? translation : reply}</MessageBox>
              )}
            </div>
          ))}
        </motion.div>
      )}
    </AnimatePresence>
  );
}
