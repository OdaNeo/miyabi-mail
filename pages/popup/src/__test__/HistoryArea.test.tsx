import { describe, it, expect, vi, beforeEach } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import { HistoryArea } from '../feature/HistoryArea';
import { useLiveQuery } from 'dexie-react-hooks';
import { useI18n } from '@/hooks/useI18n';
import { TaskHistory } from '@/db/model';
import React from 'react';
import { deleteTaskRecordById } from '@/db/taskRecordsRepository';

vi.mock('dexie-react-hooks', () => ({
  useLiveQuery: vi.fn(),
}));

vi.mock('@/db/taskRecordsRepository', () => ({
  deleteTaskRecordById: vi.fn(),
}));

vi.mock('@/hooks/useI18n', () => ({
  useI18n: vi.fn(),
}));

vi.mock('framer-motion', () => ({
  AnimatePresence: ({ children }: { children: React.ReactNode }) => <>{children}</>,
  motion: {
    div: ({ children, ...props }: any) => (
      <div data-testid="motion-div" {...props}>
        {children}
      </div>
    ),
  },
}));

describe('HistoryArea 组件', () => {
  const mockI18n = {
    NO_HISTORY: '没有历史记录',
    HISTORY_IS_HERE: '历史记录将显示在这里',
  };

  const mockHistoryItems: TaskHistory[] = [
    {
      id: 1,
      sourceContent: '测试内容1',
      targetLanguage: 'en',
      taskType: 'TRANSLATION',
      modelVersion: 'gpt-4o',
      modelTemperature: 0.7,
      generatedContent: '翻译结果1',
      createdAt: new Date('2023-01-01T10:00:00'),
      completedAt: new Date('2023-01-01T10:01:00'),
      isVisitable: true,
    },
    {
      id: 2,
      sourceContent: '测试内容2',
      targetLanguage: 'zh',
      taskType: 'REPLY',
      modelVersion: 'gpt-3.5-turbo',
      modelTemperature: 0.5,
      emailSubject: '测试邮件',
      generatedContent: '回复结果2',
      createdAt: new Date('2023-01-02T11:00:00'),
      completedAt: new Date('2023-01-02T11:02:00'),
      isVisitable: true,
    },
  ];

  beforeEach(() => {
    vi.clearAllMocks();
    (useI18n as any).mockReturnValue(mockI18n);
  });

  it('应该正确渲染历史记录列表', () => {
    (useLiveQuery as any).mockReturnValue(mockHistoryItems);

    render(<HistoryArea />);

    expect(screen.getAllByTestId('history-item')).toHaveLength(2);

    expect(screen.getByText('测试内容1')).toBeInTheDocument();
    expect(screen.getByText('翻译结果1')).toBeInTheDocument();
    expect(screen.getByText('测试内容2')).toBeInTheDocument();
    expect(screen.getByText('回复结果2')).toBeInTheDocument();

    expect(screen.getByText('2023/1/1 10:01:00')).toBeInTheDocument();
    expect(screen.getByText('2023/1/2 11:02:00')).toBeInTheDocument();

    expect(screen.getByText('翻译')).toBeInTheDocument();
    expect(screen.getByText('回复')).toBeInTheDocument();
  });

  it('应该在无历史记录时显示空状态', () => {
    (useLiveQuery as any).mockReturnValue([]);

    const { container } = render(<HistoryArea />);
    expect(container).toMatchSnapshot();
  });

  it('应该调用删除函数当点击删除按钮', async () => {
    (useLiveQuery as any).mockReturnValue(mockHistoryItems);

    render(<HistoryArea />);

    const deleteButtons = screen.getAllByTestId('delete-history-icon');
    fireEvent.click(deleteButtons[0]);
    expect(deleteTaskRecordById).toHaveBeenCalledWith(1);
  });
});
