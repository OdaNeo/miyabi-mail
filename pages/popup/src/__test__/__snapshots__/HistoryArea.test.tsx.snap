// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HistoryArea 组件 > 应该在无历史记录时显示空状态 1`] = `
<div>
  <div
    class="relative overflow-hidden scrollbar-custom p-5"
    data-testid="history-area"
    dir="ltr"
    style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
  >
    <style>
      [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
    </style>
    <div
      class="h-full w-full rounded-[inherit]"
      data-radix-scroll-area-viewport=""
      style="overflow-x: hidden; overflow-y: scroll;"
    >
      <div
        style="min-width: 100%; display: table;"
      >
        <div
          class="space-y-3"
        >
          <div
            animate="[object Object]"
            class="flex flex-col items-center justify-center py-10 text-[#6b7280] dark:text-[#9ca3af]"
            data-testid="motion-div"
            initial="[object Object]"
          >
            <svg
              class="lucide lucide-circle-alert w-10 h-10 mb-3 opacity-50"
              data-testid="history-alert-circle"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <line
                x1="12"
                x2="12"
                y1="8"
                y2="12"
              />
              <line
                x1="12"
                x2="12.01"
                y1="16"
                y2="16"
              />
            </svg>
            <p
              class="text-sm"
            >
              没有历史记录
            </p>
            <p
              class="text-sm mt-1 max-w-xs text-center"
            >
              历史记录将显示在这里
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
